### ISSUE TITLE
[Bug] SimpleGatedLinearAttention masks v for padding but not the decay gate gk

### ISSUE THREAD

**Issue by AmirF194** [fla-org/flash-linear-attention#1272]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

`SimpleGatedLinearAttention.forward` masks `v` with `attention_mask` before it reaches `chunk_simple_gla` / `fused_recurrent_simple_gla`, but not `gk` (the per-head decay gate):

```python
if attention_mask is not None:
    v = v.mul_(attention_mask[:, -v.shape[-2]:, None])
...
gk = F.logsigmoid(gk) / self.gate_logit_normalizer
```

The recurrence is `S = S * exp(gk) + k (x) v`. Zeroing `v` at a padded step drops its own contribution to `S`, but `gk` at that step is still whatever `gk_proj(hidden_states)` produced from the padding content, so `S` still gets multiplied by a real, content-derived decay factor instead of being passed through unchanged. That corrupts every later token in the same forward call once the padding is interior rather than leading, and it corrupts the cached `recurrent_state` carried into the next call if the padded step lands during a warm cache continuation.

### Steps to Reproduce the Bug

Two forward calls that are identical except for what the padded span happens to embed to (mask=0 there) diverge after the pad, on current `main` (`954438d1`):

```python
import torch
from fla.layers.simple_gla import SimpleGatedLinearAttention

torch.manual_seed(42)
B, T, H, D, PAD = 2, 128, 4, 64, 3
layer = SimpleGatedLinearAttention(hidden_size=H * D, num_heads=H, use_short_conv=False, layer_idx=0).eval()

hidden_states = torch.randn(B, T, H * D)
varied = hidden_states.clone()
varied[:, 10:10 + PAD] = torch.randn(B, PAD, H * D)  # only the padded span differs
mask = torch.ones(B, T, dtype=torch.long)
mask[:, 10:10 + PAD] = 0

with torch.no_grad():
    out_a, *_ = layer(hidden_states=hidden_states, attention_mask=mask)
    out_b, *_ = layer(hidden_states=varied, attention_mask=mask)

print((out_a[:, 10 + PAD:] - out_b[:, 10 + PAD:]).abs().max())  # should be 0, is not
```

I ran this on a CPU-only box by monkeypatching `chunk_simple_gla`/`fused_recurrent_simple_gla` to `fla.ops.simple_gla.naive`'s reference implementations and `RMSNorm`/swish to plain-PyTorch equivalents (this repo's Triton kernels need a GPU I don't have); the divergence disappears once `gk` is masked the same way `v` is, and the naive-recurrence math confirms the mechanism above.

### Expected Behavior

Output at any token after a padded span should not depend on what the padding tokens happened to embed to, whether the pad is interior to one call or a warm cache carries it across calls.

### Environment Information

1. Torch: 2.9.1+cpu (issue is in `fla/layers/simple_gla.py` Python logic, not kernel-specific)
2. Triton: n/a to the repro (kernels bypassed via the repo's own naive reference)
